src/Common/src/System/IO/StringParser.cs

```diff
@@ -171,17 +171,12 @@ public unsafe int ParseNextInt32()
                     {
                         ThrowForInvalidData();
                     }
-                    result = checked((result * 10) + d);
+                    result = negative ? checked((result * 10) - d) : checked((result * 10) + d);
```

Comment:
IMO, it makes the card harder to read and reason about.  It also introduces an extra branch on each iteration of the loop.  Up to you, though.